src/Utilities/TupleSlice.hpp

```diff
@@ -0,0 +1,54 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+/// \file
+/// Defines functions for slicing tuples
+
+#pragma once
+
+#include <cstddef>
+#include <tuple>
+#include <utility>
+
+namespace tuple_impl_detail {
+
+template <size_t Start, typename Tuple, size_t... Is>
+constexpr auto slice_impl(Tuple&& tuple,
```

Comment:
why not use trailing return type syntax instead of `auto`?

Reply:
Do you think d029f28 is better?

Reply:
Yes. I find code easier to read if it has the return type explicit. I generally dislike auto in return types because the rules are stupidly complicated for when it gives references, pointers, const, etc. I think being explicit whenever it is easy to do so is better :)